irohad/multi_sig_transactions/gossip_propagation_strategy.hpp

```diff
@@ -18,36 +18,53 @@
 #ifndef IROHA_GOSSIP_PROPAGATION_STRATEGY_HPP
 #define IROHA_GOSSIP_PROPAGATION_STRATEGY_HPP
 
-#include <rxcpp/rx.hpp>
+#include "ametsuchi/peer_query.hpp"
+#include "multi_sig_transactions/mst_propagation_strategy.hpp"
+#include <memory>
 #include <queue>
-#include "mst_propagation_strategy.hpp"
+#include <rxcpp/rx.hpp>
 
 namespace iroha {
 
-  /**
-   * Interface provides strategy for propagation states in network
+/**
+ * This class provides strategy for propagation states in network
+ * It makes randomly ordered queue of peers and then fetch until
+ * the queue is empty on the provided period.
+ */
+class GossipPropagationStrategy : public PropagationStrategy {
+public:
+  /*
+   * Initialize strategy with:
+   * @param query that provider list of peers
+   * @param period of emitting to observable in ms
+   * @param amount of peers emitted per once
+   */
+  GossipPropagationStrategy(std::shared_ptr<ametsuchi::PeerQuery> query,
+                            uint32_t period, uint32_t amount);
```

Comment:
Maybe better to change period with `chrono`.

Reply:
yeah, good point, i'll fix that